Modules/Loadable/SubjectHierarchy/Widgets/qMRMLSortFilterSubjectHierarchyProxyModel.cxx

```diff
@@ -198,6 +198,14 @@ QModelIndex qMRMLSortFilterSubjectHierarchyProxyModel::indexFromSubjectHierarchy
   return this->mapFromSource(sceneModel->indexFromSubjectHierarchyItem(itemID, column));
 }
 
+//-----------------------------------------------------------------------------
+QStandardItem* qMRMLSortFilterSubjectHierarchyProxyModel::itemFromSubjectHierarchyItem(vtkIdType itemId, int column)const
+{
+  qMRMLSubjectHierarchyModel* sceneModel = qobject_cast<qMRMLSubjectHierarchyModel*>(this->sourceModel());
+  QModelIndex index = this->mapFromSource(sceneModel->indexFromSubjectHierarchyItem(itemId, column));
+  return sceneModel->itemFromIndex(index);
```

Comment:
@cpinter is that what you were envisioning? This currently returns a null pointer in my use case for every item [here](https://github.com/Slicer/Slicer/pull/813/files#diff-784aa1951ed8cd51bfe8a1bb9fe2ee5bR628), so i'll keep investigating.

Reply:
First of all, please don't reimplement anything that is there and works, i.e.
https://github.com/Slicer/Slicer/blob/master/Modules/Loadable/SubjectHierarchy/Widgets/qMRMLSortFilterSubjectHierarchyProxyModel.cxx#L195

About the actual issue, I'm not sure if using the scene model's itemFromIndex function on an index coming form the proxy model would work. I'll take a deeper look.

Reply:
This call works for the highlights
https://github.com/Slicer/Slicer/blob/master/Modules/Loadable/SubjectHierarchy/Widgets/qMRMLSubjectHierarchyTreeView.cxx#L1247
I wonder why it didn't work for you. Apparently the scene model can be used to get the items of the columns from SH ID. I'd start from here. Thanks!